### PR TITLE
AI based heuristic for Find 

### PR DESCRIPTION
**Pull request by JehandadKhan**
Testing in progress

### CODE REVIEW

File: src/conv/heur/metadata.cpp [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+  const auto &arch = handle.GetDeviceName();
+  auto fwd = problem.direction.IsForward();
+  auto bwd = problem.direction.IsBackwardData();
+  auto wrw = problem.direction.IsBackwardWrW();
+
+  if (arch == "gfx908" && fwd)
+
+  {
+    static const std::vector<float> gfx908_fwd_mu = {
+        3.5623828968151168, 1.0490354202886407, 1.0490354202886407,
```
Comment (asleepzzz) [inline src/conv/heur/metadata.cpp:198]: WHAT'S THIS

**Comment by atamazov** [comment]
@JehandadKhan I've reworked the most of the `imm-fallback-improv` changes in another branch, `getwti-wino2111`, see https://github.com/ROCmSoftwarePlatform/MIOpen/compare/imm-fallback-improv...getwti-wino2111.

`getwti-wino2111` implements `GetWti()` for GEMM and Winograd and takes it into use. I am going to finalize `getwti-wino2111` & open PR when #424 is merged in.

Can you rebase this PR on top of `getwti-wino2111`? Or you can simply add the new code, and I will integrate is for you in separate PR. What do you think?

File: src/ocl/convolutionocl.cpp [hunk 2]
```diff
@@ -1583,7 +1584,66 @@ void ConvolutionDescriptor::GetSolutionsFallback(Handle& handle,
...
+                ws_sz = WrwGetValidWorkSpaceSizeGemm(outDesc, inDesc, weightsDesc);
+            }
+            else
+                continue; // Its not applicable
+            interim.emplace_back(-1, ws_sz, solver::Id::gemm().Value(), miopenConvolutionAlgoGEMM);
+        }
+        else
+        {
+            const auto sol = solver_id.GetSolver();
+            if(!sol.IsApplicable(ctx))
```
Comment (atamazov) [inline src/ocl/convolutionocl.cpp:1618]: I believe you wanted to put this after IsDynamic check ;)

File: src/conv/heur/metadata.cpp (hunk 1 above)
Comment (JehandadKhan) [reply src/conv/heur/metadata.cpp:198]: These are the mean and standard deviation vectors for the features from the dataset. Which we need to normalize the features everytime we use the model to do inference.

**Comment by JehandadKhan** [comment]
@atamazov  I created a new PR #608 which supersedes this PR and is based off your getwti-wino2011 branch as requested.

**Comment by atamazov** [comment]
@JehandadKhan Thank yo very much!

Unfortunately more changes in baseline of this PR will come. The reason is that `imm-fallback-improv` is not merged yet and therefore I am unable to finalize `getwti-wino2011` (and create a PR for it).

**Pull request closed without merging**